dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/domain/batching/processors/DefaultDataProcessorTest.kt

```diff
@@ -69,6 +98,33 @@ internal class DefaultDataProcessorTest {
         val argumentCaptor = argumentCaptor<Runnable>()
         verify(mockedExecutorService).submit(argumentCaptor.capture())
         argumentCaptor.firstValue.run()
-        verify(mockedWriter).write(fakeEvents)
+        fakeEvents.forEach {
+            verify(mockedWriter).write(it)
+        }
+    }
+
+    @Test
+    fun `M do nothing when event mapper returns null for an event W a list of events is consumed`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val fakeEvents = forge.aList { forge.aString() }
+        val fakeRandIndex = forge.anInt(0, fakeEvents.size)
+        whenever(mockedEventMapper.map(fakeEvents[fakeRandIndex])).thenReturn(null)
+
+        // WHEN
+        testedDataProcessor.consume(fakeEvents)
+
+        // THEN
+        val argumentCaptor = argumentCaptor<Runnable>()
+        verify(mockedExecutorService).submit(argumentCaptor.capture())
+        argumentCaptor.firstValue.run()
+        fakeEvents.forEachIndexed() { index, event ->
+            if (index != fakeRandIndex) {
+                verify(mockedWriter).write(event)
+            } else {
+                verify(mockedWriter, never()).write(event)
+            }
+        }
     }
```

Comment:
What if the mapper returns an event that is different from the input ? We should define the behavior to use and add tests accordingly.

Reply:
Hmm good point. I am thinking what should we do in that case ? Maybe we should drop the event.